lib/daru/io/exporters/json.rb

```diff
@@ -0,0 +1,403 @@
+        #   #            name          age       sex
+        #   # child   Jon Snow         18       Male
+        #   # dad   Rhaegar Ta         54       Male
+        #   # mom   Lyanna Sta         36     Female
+        #
+        #   Daru::IO::Exporters::JSON.new(
+        #     df,
+        #     'filename.json',
+        #     orient: :records,
+        #     pretty: true,
+        #     name: '$..specific..name',
+        #     age: '$..common..age',
+        #     sex: '$..common..gender'
+        #   )
+        #
+        #   #=>
+        #   # [
+        #   #   {
+        #   #     "common": {
+        #   #       "gender": "Male",
+        #   #       "age": 18
+        #   #     },
+        #   #     "specific": {
+        #   #       "name": "Jon Snow"
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "common": {
+        #   #       "gender": "Male",
+        #   #       "age": 54
+        #   #     },
+        #   #     "specific": {
+        #   #       "name": "Rhaegar Targaryen"
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "common": {
+        #   #       "gender": "Female",
+        #   #       "age": 36
+        #   #     },
+        #   #    "specific": {
+        #   #       "name": "Lyanna Stark"
+        #   #     }
+        #   #   }
+        #   # ]
+        # @example Writing to a JSON file with dynamic JsonPaths
+        #   df = Daru::DataFrame.new(
+        #     [
+        #       {name: 'Jon Snow', age: 18, sex: 'Male'},
+        #       {name: 'Rhaegar Targaryen', age: 54, sex: 'Male'},
+        #       {name: 'Lyanna Stark', age: 36, sex: 'Female'}
+        #     ],
+        #     order: %i[name age sex],
+        #     index: %i[child dad mom]
+        #   )
+        #
+        #   #=> #<Daru::DataFrame(3x3)>
+        #   #            name          age       sex
+        #   # child   Jon Snow         18       Male
+        #   # dad   Rhaegar Ta         54       Male
+        #   # mom   Lyanna Sta         36     Female
+        #
+        #   Daru::IO::Exporters::JSON.new(
+        #     df,
+        #     'filename.json',
+        #     orient: :records,
+        #     pretty: true,
+        #     age: '$..{name}..age',
+        #     sex: '$..{name}..gender'
+        #   )
+        #
+        #   #=>
+        #   # [
+        #   #   {
+        #   #     "Jon Snow": {
+        #   #       "gender": "Male",
+        #   #       "age": 18
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "Rhaegar Targaryen": {
+        #   #       "gender": "Male",
+        #   #       "age": 54
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "Lyanna Stark": {
+        #   #     "gender": "Female",
+        #   #     "age": 36
+        #   #     }
+        #   #   }
+        #   # ]
+        def initialize(dataframe, path, orient: :index, pretty: false, **jsonpaths)
```

Comment:
The `:orient` option is inspired from [pandas](https://github.com/pandas-dev/pandas/blob/v0.20.3/pandas/core/generic.py#L1129-L1245), which can be set to `:records`, `:split`, `:values` and defaults to `:index`.